crates/llama-cpp-bindings/src/llama.rs

```diff
@@ -1,104 +1,87 @@
-use std::{collections::HashMap, thread::JoinHandle};
+use std::{thread::JoinHandle};
 
 use cxx::UniquePtr;
-use tokio::sync::mpsc::{channel, Receiver, Sender};
+use tokio::sync::{
+    mpsc::{channel, unbounded_channel, Receiver, Sender, UnboundedReceiver, UnboundedSender},
+    RwLock,
+};
 
 use crate::ffi;
 
-struct LlamaInitRequest {
+pub(crate) struct LlamaInitRequest {
+    id: u32,
     prompt: String,
     max_input_length: usize,
     temperature: f32,
     seed: u64,
 
-    tx: Sender<String>,
+    tx: UnboundedSender<String>,
 }
 
-struct LlamaRunningRequest {
-    tx: Sender<String>,
+impl LlamaInitRequest {
+    pub(crate) fn id(&self) -> u32 {
+        self.id
+    }
+
+    pub(crate) fn prompt(&self) -> &str {
+        &self.prompt
+    }
+
+    pub(crate) fn max_input_length(&self) -> usize {
+        self.max_input_length
+    }
+
+    pub(crate) fn seed(&self) -> u64 {
+        self.seed
+    }
+
+    pub(crate) fn temperature(&self) -> f32 {
+        self.temperature
+    }
+
+    pub(crate) fn check_candidate(&self, _candidate: &[u8]) -> bool {
+        true
+    }
+
+    pub(crate) fn accept_candidate(&self, _candidate: &[u8]) {}
```

Comment:
@boxbeam for now they're dummy implementation, but should provide sufficient flexibility to integrate a treesitter context